### DISCUSSION TITLE
Ignore typescript type errors in `.js` / `.jsx` files

### DISCUSSION THREAD

**Discussion by AgrYpn1a** [typescript-language-server/typescript-language-server#905] [Q&A]
I am using Emacs and Eglot to interact with the typescript-language-server by invoking the standard command. The project that I am working on is a React project that uses plain JS, and I want the .js files to be checked in order to be able to detect and fix missing imports, unused variables, autocomplete and similar. However, it is also reporting typescript errors and those also complain about JSX components being invalid which is coloring a bunch of stuff in red in source files which is distracting. 

I went through the configuration docs and I could not figure out how to tell the server to not check for types and type errors in the .js files.

**Comment by x4132** [comment]
set the property `checkJs` to `false` in the compilerOptions in your tsconfig.json, this is not a issue with tsserver.

**Comment by AgrYpn1a** [comment]
@x4132 If I do so I will lose code completion, import auto fixes, unused variable detection and so on which is then the same if I didn't have the lsp on.

**Comment by rchl** [comment]
`tsconfig.json` is generally where you can configure such things. See full reference at https://www.typescriptlang.org/tsconfig/

I'm not sure what errors exactly do you want and not want to see so hard to be more specific with the answer.

**Reply by AgrYpn1a** [reply]
Basically, I want to have auto-complete, import fixes and maybe some other code actions (quickfixes) and am not interested in any Typescript type related stuff, since I am working on a plain JS project. The errors I am having are for example:

```
21  82 error    e-f-b    typescript [2554]: Expected 0 arguments, but got 1.
   31  13 error    e-f-b    typescript [2786]: 'AppLayout' cannot be used as a JSX component.
   34  25 error    e-f-b    typescript [2786]: 'Skeleton' cannot be used as a JSX component.
   35  29 error    e-f-b    typescript [2786]: 'Typography' cannot be used as a JSX component.
   38  25 error    e-f-b    typescript [2786]: 'Box' cannot be used as a JSX component.
   38  25 error    e-f-b    typescript [2769]: No overload matches this call.
   48  29 error    e-f-b    typescript [2786]: 'CircularProgress' cannot be used as a JSX component.
   59  16 error    e-f-b    typescript [2769]: No overload matches this call.
   62  13 error    e-f-b    typescript [2786]: 'NotificationsContainer' cannot be used as a JSX component.
   63  13 error    e-f-b    typescript [2786]: 'ErrorBoundary' cannot be used as a JSX component.
   64  17 error    e-f-b    typescript [2786]: 'AppContent' cannot be used as a JSX component.
   65  21 error    e-f-b    typescript [2786]: 'RouteElements' cannot be used as a JSX component.
```
